colossalai/booster/plugin/moe_hybrid_parallel_plugin.py

```diff
@@ -323,7 +339,10 @@ def prepare_dataloader(
         """
         _kwargs = kwargs.copy()
         sampler = DistributedSampler(
-            dataset, num_replicas=self.pg_mesh.size(DP_AXIS), rank=self.pg_mesh.coordinate(DP_AXIS), shuffle=shuffle
+            dataset,
+            num_replicas=self.pg_mesh.size(self.dp_axis),
+            rank=self.pg_mesh.coordinate(self.dp_axis),
```

Comment:
Should we use global dp rank and size?

Reply:
yes, we should use global dp process group's rank/size to construct the distributed sampler